src/modules/expression/macros.rs

```diff
@@ -0,0 +1,259 @@
+#[macro_export]
+macro_rules! parse_operator {
+    ($meta:expr, $start_index:expr, $name: ident $block:tt) => {{
+        use heraclitus_compiler::prelude::*;
+        let mut module = $name $block;
+        syntax($meta, &mut module)?;
+        let end_index = $meta.get_index();
+        Expr {
+            kind: module.get_type(),
+            value: Some (ExprType:: $name (module)),
+            pos: ($start_index, end_index)
+        }
+    }}
+}
+
+#[macro_export]
+macro_rules! parse_expr_group {
+    // Group type that handles Binary Operators
+    (@internal ({$cur:ident, $prev:ident}, $meta:expr, BinOp => [$($cur_modules:ident),+])) => {{
```

Comment:
I get what you're doing here, but I'm not sure if it's the right direction.

Right now, there are macros just for expressions. What about statements?

Import statement is vastly different than loop for example. Do we want macros just for one type of expr/stmnt like is the case with ternary? Or do we just parse them "manually" without macro?

Reply:
The thing is that binary operators and other operators have to be parsed differently than other syntax modules. Statements are fine with what there is already. There is no statement-precedence as they are evaluated line by line. So there is no need to do that. The change is just for binary, ternary and unary operators to work so that they are parsed from left to right.

I’ve done a research and this is the most popular way to parse left-right operators.